### DISCUSSION TITLE
Close terminal output window

### DISCUSSION THREAD

**Discussion by MatthiasJakobs** [mfussenegger/nvim-dap#859] [Q&A]
Is there a function I can call to programatically close the terminal output after the debugging session is over?
Currently, I have a function set up that closes the REPL and terminates the debugging process itself, but the `dap-terminal` pane keeps open. I would like to close that as well (Currently I  "move down and send `:q` " but I would like to not specify where the output is located because I might want to change that as well later on).

Currently, I have:

```lua
local dap = require('dap')
function close_all_debug_panes()
    dap.repl.close()
    dap.disconnect({terminateDebugee = true})
end
vim.keymap.set('n', '<Leader>dq', '<cmd> lua close_all_debug_panes()<CR><C-w>j:q<CR>')
```

EDIT:
I just noticed that the terminal output is not created again when running the debugger multiple times and it was closed. So for now, I'll keep it open, but if there are any functions to open/close the output that would be great on smaller displays

**Comment by mfussenegger** [comment]
Currently there is no such functions and terminal buffers are re-used when new launching new debug sessions. You've two options:

- Set something like `dap.defaults.fallback.terminal_win_cmd = 'tabnew'` to move the terminal out of the way

- Iterate over the buffers and check their names. For example:

```lua
local bufs = vim.api.nvim_list_bufs()
for _, buf in pairs(bufs) do
  local name = vim.fn.bufname(buf)
  if vim.startswith(name, '[dap-terminal]') then
    vim.api.nvim_buf_delete(buf, {})
  end
end
```
